### PR TITLE
Minor: refine Partitioning documentation

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes #.

## Rationale for this change

I was looking for the page on [`Partitioning`](https://docs.rs/datafusion/latest/datafusion/physical_expr/enum.Partitioning.html) but the first
thing that comes up is the partitioning description for Logical plans

Screenshot 2024-08-24 at 5 27 42 AM

Thus I would like to link the description in the result that also appears first

## What changes are included in this PR?

Add a link from logical `Partitioning` to the more detailed documentation

Here is what the doc looks like now:

Screenshot 2024-08-24 at 5 30 30 AM

## Are these changes tested?

By doc CI  and I also looked at it manually

### CODE REVIEW

File: datafusion/physical-expr/src/partitioning.rs [hunk 1]
```diff
@@ -24,8 +24,8 @@ use crate::{physical_exprs_equal, EquivalenceProperties, PhysicalExpr};
 
 /// Output partitioning supported by [`ExecutionPlan`]s.
 ///
-/// When `executed`, `ExecutionPlan`s  produce one or more independent stream of
-/// data batches in parallel, referred to as partitions. The streams are Rust
+/// Calling [`ExecutionPlan::execute`] produce one or more independent streams of
```
Comment (alamb) [inline datafusion/physical-expr/src/partitioning.rs:27]: This was a driveby cleanup

File: datafusion/expr/src/logical_plan/plan.rs [hunk 2]
```diff
@@ -2927,7 +2927,11 @@ impl Debug for Subquery {
     }
 }
 
-/// Logical partitioning schemes supported by the repartition operator.
+/// Logical partitioning schemes supported by [`LogicalPlan::Repartition`]
+///
+/// See [`Partitioning`] for more details on partitioning
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:2932]: the main reason for this PR is to add this link

File: datafusion/physical-expr/src/partitioning.rs [hunk 3]
```diff
@@ -24,8 +24,8 @@ use crate::{physical_exprs_equal, EquivalenceProperties, PhysicalExpr};
...
+/// [`RecordBatch`]es in parallel, referred to as partitions. The streams are Rust
 /// `async` [`Stream`]s (a special kind of future). The number of output
```
Comment (comphead) [inline datafusion/physical-expr/src/partitioning.rs:29]: perhaps we can add the link to the `futures-core` crate which backs streams ?

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: datafusion/physical-expr/src/partitioning.rs (hunk 3 above)
Comment (alamb) [reply datafusion/physical-expr/src/partitioning.rs:29]: I double checked by building the docs locally

This link looks like this:
Screenshot 2024-08-25 at 6 59 37 AM


And it goes to this link:
https://docs.rs/futures/latest/futures/stream/trait.Stream.html

Is that what you mean by link to `futures-core`? Or do you mean something else?

Comment (comphead) [reply datafusion/physical-expr/src/partitioning.rs:29]: exactly, my bad, when I navigated from IDE it showed me the crate name as `futures-core` sub crate which is confusing

**Comment by alamb** [comment]
Thanks @comphead
